modules/java/android_sdk/android_gradle_lib/build.gradle

```diff
@@ -9,12 +9,11 @@ android {
         targetSdkVersion @ANDROID_TARGET_SDK_VERSION@
     }
 
+    packagingOptions {
+        doNotStrip '**/*.so'  // Strip is controlled by OpenCV CMake scripts
+    }
+
     buildTypes {
-        debug {
-            packagingOptions{
-                doNotStrip '*.so'
-            }
-        }
         release {
```

Comment:
The exact working of `packagingOptions` and `doNotStrip` are somewhat of a mystery but I was never able to get them to work outside of the `buildTypes` block. If you want to specify `doNotStrip` for both `debug` and `release` I think the following would work:
```
    buildTypes {
        debug {
            packagingOptions {
                doNotStrip '**/*.so'
            }
        }
        release {
            packagingOptions {
                doNotStrip '**/*.so'
            }
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.txt'
        }

```